### PR TITLE
Feature findings api enhancements

### PR DESCRIPTION
**Pull request by riysaxen-amzn**
### Description
This PR is related to paginations enhancements, for more info please refer to the below github issue
 
### Issues Resolved
[795](https://github.com/opensearch-project/security-analytics/issues/795)]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by sbcd90 — APPROVED** [review]

File: src/main/java/org/opensearch/securityanalytics/transport/TransportGetFindingsAction.java [hunk 1]
```diff
@@ -103,66 +106,88 @@ protected void doExecute(Task task, GetFindingsRequest request, ActionListener<G
...
+                    ),
+                    ScoreMode.None
+            );
+            searchSourceBuilder.query(queryBuilder);
+        }
+        else {
+            MatchAllQueryBuilder queryBuilder = QueryBuilders.matchAllQuery();
+            searchSourceBuilder.query(queryBuilder);
         }
+        searchSourceBuilder.fetchSource(true);
```
Comment (eirsep) [inline src/main/java/org/opensearch/securityanalytics/transport/TransportGetFindingsAction.java:185]: if we dont pass size, it defaults to 10

Comment (eirsep) [reply src/main/java/org/opensearch/securityanalytics/transport/TransportGetFindingsAction.java:185]: won't consider more than 10 detector's findings

File: src/test/java/org/opensearch/securityanalytics/findings/FindingIT.java [hunk 2]
```diff
@@ -278,6 +272,582 @@ public void testGetFindings_byDetectorType_success() throws IOException {
         Assert.assertEquals(1, getFindingsBody.get("total_findings"));
     }
 
+    public void testGetAllFindings_success() throws IOException {
```
Comment (eirsep) [inline src/test/java/org/opensearch/securityanalytics/findings/FindingIT.java:275]: does this test cover edge cases. why is this a big bang method and not smaller methods testing various scenarios?

File: src/main/java/org/opensearch/securityanalytics/transport/TransportGetFindingsAction.java (hunk 1 above)
Comment (riysaxen-amzn) [reply src/main/java/org/opensearch/securityanalytics/transport/TransportGetFindingsAction.java:184]: yes, make sense. I had this earlier, i removed it in mid while developing this. Added back

File: src/test/java/org/opensearch/securityanalytics/findings/SecureFindingRestApiIT.java [hunk 3]
```diff
@@ -31,12 +31,7 @@
 import java.util.Map;
 import java.util.stream.Collectors;
 
-import static org.opensearch.securityanalytics.TestHelpers.netFlowMappings;
-import static org.opensearch.securityanalytics.TestHelpers.randomDetectorType;
-import static org.opensearch.securityanalytics.TestHelpers.randomDetectorWithTriggers;
-import static org.opensearch.securityanalytics.TestHelpers.randomDoc;
-import static org.opensearch.securityanalytics.TestHelpers.randomIndex;
-import static org.opensearch.securityanalytics.TestHelpers.windowsIndexMapping;
+import static org.opensearch.securityanalytics.TestHelpers.*;
```
[the remark this answers is not in the document]
Comment (riysaxen-amzn) [reply src/test/java/org/opensearch/securityanalytics/findings/SecureFindingRestApiIT.java]: addressed

File: src/test/java/org/opensearch/securityanalytics/findings/FindingIT.java (hunk 2 above)
Comment (riysaxen-amzn) [reply src/test/java/org/opensearch/securityanalytics/findings/FindingIT.java:275]: This test covers the case to get all the detector findings on a rule match irrespective of logType/detectortType basically this API  `_security_analytics/findings/_search?` We need to refactor this entire class actually, there are many methods like this, we can create a seperate backlog for the integ tests refactoring

**Review by eirsep — DISMISSED** [review]

File: src/main/java/org/opensearch/securityanalytics/findings/FindingsService.java [hunk 4]
```diff
@@ -22,6 +23,7 @@
 import org.opensearch.commons.alerting.model.FindingWithDocs;
 import org.opensearch.commons.alerting.model.Table;
 import org.opensearch.core.rest.RestStatus;
+import org.opensearch.index.query.*;
 import org.opensearch.securityanalytics.action.FindingDto;
```
[the remark this answers is not in the document]
Comment (riysaxen-amzn) [reply src/main/java/org/opensearch/securityanalytics/findings/FindingsService.java:31]: addressed

**Review by sbcd90 — APPROVED** [review]

**Review by praveensameneni — APPROVED** [review]
